src/MediaWiki/Connection/Database.php

```diff
@@ -320,10 +323,52 @@ public function select( $tableName, $fields, $conditions, $fname, array $options
 	 * @throws RuntimeException
 	 */
 	public function query( $sql, $fname = __METHOD__, $ignoreException = false ) {
-
-		$connection = $this->connRef->getConnection( 'write' );
 		$this->transactionHandler->muteTransactionProfiler( true );
 
+		$results = $this->executeQuery(
+			$this->connRef->getConnection( 'write' ),
+			$sql,
+			$fname,
+			$ignoreException
+		);
+
+		$this->transactionHandler->muteTransactionProfiler( false );
```

Comment:
As a side-effect, this infrastructure can go away as well once we remove the use of master connections from read requests :)